### ISSUE TITLE
Suggest improvement "Build question form" UI

### ISSUE THREAD

**Issue by nagthanm** [fit-together-tv/ft-issue-report#9]
### Reproduction Steps
User steps
* Open [app.chuongnau.com](https://app.chuongnau.com/me)
* Choose "Add new form" or "View question form"
* Create Questions/ Build your question form
* **"Save questions"**
* Choose "Publish Test and answer"
* "Create exam" -> **"Copy exam link"**
* Start the exam
  
Participant steps
* Open the **"Copy exam link"** which is provided by user.
* Start the exam
* Finish the exam
* "Submit answer"

### User experience
* Reproduction Steps continue:
    * User Steps
      * Clients tend to continue the work from "Build question form" tab, because most app do not need to go back to "Profile" page to create new question form.
      * Consequences, they tend to save new question form on existed question form.  And "Save questions" button have no warning note that the form can be change or ask you to save with a different name.
      * Change "Form name" directly on the "build question form" tab do not change the question form path.
   * Participants steps
      * Therefore, participants will received the same exam path (same question form path) which they have already done but with different questions and question groups. They "submit answer" and the outcome -> "You already have an existing submission for this question form."

### Suggest Improvement
*On tab "Build question form", Should have a tab "File" like most application: New File, Open File, Save File, Save As..., Export, Import, etc. 
* Click "Save Question" and have warning: The form was changed. Do you wish to save on the existing form or save as new form.
* Change "Form name" and also a warning should pop up: Do you wish to change to different "Question form" path or you just need to change the "form name".

### Device:
I have just encountered this experince on PC Windows 10.

### Browser:
N/A

![You already have an existing submission for this question form](https://user-images.githubusercontent.com/87641533/127745943-cac7eaa0-4f0b-4055-9d23-f35121a041c0.png)

**Comment by bugless-it** [comment]
Actually this is a design i thought of in the past, but after talking to users, they mentioned that they wanted to ability to change the question forms like this, since their workflow required frequent corrections. They also mentioned that they wouldn't change the question form after students have taken the tests.

**Comment by nagthanm** [comment]
Could you rephrase it? :))
I do not really understand what's your plan for this issue?

**Comment by bugless-it** [comment]
so the timeline was like this:
- Originally, I had a similar design whereby teachers are not allowed to change a question form once it has active tests.
- A teacher told me that it was inconvenient for her, since she needed to do a lot of back-and-forth collaboration with other people before settling on the final questions - e.g. create, save, send for testing, then edit again, multiple times.
- She also mentioned that teachers would not change the questions once they had finalized. So adding a new popup may slow them down.

**Comment by nagthanm** [comment]
Ok, I got your point and their point now. 
Do you think users will remember to go back to profile page to create a new question or they will mistaken delete their question group and create a new questions or question group on an existing question form.
Because I do not find most app need to open new file on profile page.
Perhaps a function like "save and exit" or "save and open new file" should be added.
@bugless-it

**Comment by bugless-it** [comment]
Let me think about this. Maybe we can add a button called `save as a new form`?

**Comment by nagthanm** [comment]
I suppose `save as a new form` is save the already existed question form with different name and different path, ain't I?
I think we can make a downward triangle icon or `^` but point downward right next to `save question` as a sub-function . Click on the icon and the user can have option to `save and build new question form` or `save as a new form`.
what do you think about that?

**Comment by bugless-it** [comment]
@nagthanm can you elaborate on the difference between `save as a new form` and `save and build a new question form`?

**Comment by nagthanm** [comment]
+ Hi 'save and build a new question form' means the user will save and completely close the previous form and open a blank page for a fresh start.
+ 'Save as a new form', I suppose That you save your ongoing form with different name, in case you want to have different versions of your ongoing form.

**Comment by bugless-it** [comment]
`Save as a new form` will create another form, just with the same content, so changes to the new do not affect the old form. For the `save and build new question form`, i think it might be better if i add a button on the top app bar like this:

![Capture](https://user-images.githubusercontent.com/52926176/128619664-6c84b145-d4ae-47cf-8406-700e483dba53.PNG)

so users can quickly create a new form. What do you think?

**Comment by nagthanm** [comment]
* Hi, that't one good way to solve the issue. I might need time to experience and give you the final answer.
* And where would you set up 'Save as a new form' .

**Comment by bugless-it** [comment]
I'm thinking of adding a button right below `Save question form`, with the title `Save as new question form`.

**Comment by bugless-it** [comment]
@nagthanm I was thinking of adding `Save as a new form` right below `Save question form`.
